### PR TITLE
Make doc structure more user friendly

### LINKED ISSUE #3096: Make docs home page friendlier
### Problem
Currently the landing page of the docs is not super inviting. https://ipywidgets.readthedocs.io/en/latest/index.html#
image

1. There is no explanatory or welcome text
2. The TOC in the sidebar is totally collapsed:
image

### Proposed Solution
1. Add some friendly opening text.
    - e.g. "Ipywidgets is a library that ...." - "If you want to learn ___ go to `<link>`, if you want to ___ go to `<link 2>`"
2. Add a gif of some awesome widget interactions
     - and/or add some active widgets to the 

### Additional Context:
Two widget libraries that could serve as inspiration:
https://ipycytoscape.readthedocs.io/en/latest/
https://ipycanvas.readthedocs.io/en/latest/

### PR DESCRIPTION
**Pull request by willingc**
Closes #3096 

- Edit landing page:
    - Add version and date
    - Add intro paragraph about ipywidgets
    - Update the table of contents into subsections with captions
- Reorder where `Using interact` falls in table of contents; it's confusing to users to place before the basic list of widgets
- Update documentation development page
- Add 7.6 content from changelog in 7.x branch here

### CODE REVIEW

**Comment by willingc** [comment]
Backport to 7.x branch after merge.

**Comment by jtpio** [comment]
Thanks!

File: docs/source/changelog.md
Comment (jtpio) [inline docs/source/changelog.md:1]:
```suggestion
# ipywidgets Changelog
```

**Comment by willingc** [comment]
Thanks for the review @jtpio :sunny:

File: docs/source/changelog.md
Comment (jasongrout) [inline docs/source/changelog.md:5]: I think this changelog entry comes in with #3086 too, so hopefully there is no conflict.

Comment (willingc) [reply docs/source/changelog.md:5]: Do you want to land #3086 now and I can rebase or do you want me to merge this now.

Comment (jasongrout) [reply docs/source/changelog.md:5]: I was giving sylvain or vidar time for one more day to look at it before merging.

Comment (willingc) [reply docs/source/changelog.md:5]: Pretty sure this is just a complete copy/paste. How adventurous do you feel?

Comment (willingc) [reply docs/source/changelog.md:5]: Except for the title.

Comment (willingc) [reply docs/source/changelog.md:5]: I'll land this after #3086.

File: docs/source/changelog.md
Comment (MSeal) [inline docs/source/changelog.md:9]: Should we change the doc string here to account for patch versions? Changing to 7.6.3 in docs would be good but if a .4 patch is ever needed it can be hard to remember to update these so maybe a `<7.7` would be best?

**Review by MSeal — APPROVED** [review]
LGTM, just the one comment around install messaging

File: docs/source/index.rst
Comment (ianhi) [inline docs/source/index.rst:18]: Perhaps remove the usage of `we` here. Per the [jupyterlab docs guidelines](https://jupyterlab.readthedocs.io/en/stable/developer/contributing.html#writing-style)

> The documentation should be written in the second person, referring to the reader as “you” and not using the first person plural “we.” The author of the documentation is not sitting next to the user, so using “we” can lead to frustration when things don’t work as expected.

Comment (ianhi) [reply docs/source/index.rst:18]: Perhaps also say something about discourse + gitter + stackoverflow tag

**Review by ianhi — COMMENTED** [review]
Huge improvement thanks!

I have one concern about language below but wouldn't hold this up on that.

File: docs/source/index.rst
Comment (willingc) [reply docs/source/index.rst:18]: Sure I can change that. It was just copy pasted from the README.

File: docs/source/index.rst
Comment (ianhi) [inline docs/source/index.rst:10]: I found it a bit jarring to be linked out github here. Though I'm not sure where (if anywhere) this link would go otherwise. It's just that github is one of the few places where there are no actually interactive html widgets for a reader to play with.

Maybe a binder link would be best?

File: docs/source/changelog.md
Comment (willingc) [reply docs/source/changelog.md:9]: I will handle this in an upcoming doc change after this PR and #3086 land. This way it's a bit easier to align content changes across both branches vs. just structural changes.

File: docs/source/index.rst
Comment (willingc) [reply docs/source/index.rst:10]: Sure. That's reasonable. This PR was simply to restructure the content structure and less focused on the content itself. I think changes like these would be great follow-up PRs.

**Comment by jasongrout** [comment]
@willingc - it looks like there is no rebase needed!

Given the replies above, I'll go ahead and merge this to get the structural changes in, and like you mention above, fine-tuning the language can be addressed in follow-up PRs. Thanks!

**Comment by jasongrout** [comment]
I made https://github.com/jupyter-widgets/ipywidgets/issues/3109 to keep track of resolving the comments above.

**Pull request merged**
